sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -1643,22 +1643,16 @@ pi_result cuda_piMemBufferCreate(pi_context context, pi_mem_flags flags,
   try {
     ScopedContext active(context);
     CUdeviceptr ptr;
-    _pi_mem::alloc_mode allocMode = _pi_mem::alloc_mode::classic;
+    _pi_mem::mem_::buffer_mem_::alloc_mode allocMode =
+        _pi_mem::mem_::buffer_mem_::alloc_mode::classic;
 
     if ((flags & PI_MEM_FLAGS_HOST_PTR_USE) && enableUseHostPtr) {
       retErr = PI_CHECK_ERROR(
           cuMemHostRegister(host_ptr, size, CU_MEMHOSTREGISTER_DEVICEMAP));
       retErr = PI_CHECK_ERROR(cuMemHostGetDevicePointer(&ptr, host_ptr, 0));
-      allocMode = _pi_mem::alloc_mode::use_host_ptr;
-    } else if (flags & PI_MEM_FLAGS_HOST_PTR_ALLOC) {
-      retErr = PI_CHECK_ERROR(cuMemAllocHost(&host_ptr, size));
-      retErr = PI_CHECK_ERROR(cuMemHostGetDevicePointer(&ptr, host_ptr, 0));
-      allocMode = _pi_mem::alloc_mode::alloc_host_ptr;
```

Comment:
I think it was deleted by accident.

Reply:
Well spotted, I am not sure why git has decided to remove these lines rather than flag it as a conflict. I see now that these are additions from #1997, which has since been merged. I will rebase this PR.

Reply:
> I think it was deleted by accident.

Marked WIP until I fix this.

Reply:
Fixed.